packages/cli/src/build-ipfs-connection.util.ts

```diff
@@ -0,0 +1,66 @@
+import IPFS from "ipfs-core"
+
+import dagJose from 'dag-jose'
+// @ts-ignore
+import multiformats from 'multiformats/basics'
+// @ts-ignore
+import legacy from 'multiformats/legacy'
+import ipfsClient from "ipfs-http-client"
+import { IpfsApi } from "@ceramicnetwork/common"
+
+multiformats.multicodec.add(dagJose)
+const dagJoseFormat = legacy(multiformats, dagJose.name)
+
+const IPFS_DHT_SERVER_MODE = process.env.IPFS_DHT_SERVER_MODE === 'true'
+
+const BOOTSTRAP_LIST = [
+    '/dns4/ipfs-dev.3boxlabs.com/tcp/4012/wss/p2p/QmeRWpCaEpQLY1yJFFHf7pFbZZapXoj5E4pKvUtyExMUKk',
+    '/dns4/ipfs-dev.ceramic.network/tcp/4012/wss/p2p/QmadvKqYyLMHYjnwswd2bHM3UCFp4F4oSFdobYUKSPjBaZ',
+    '/dns4/ipfs-dev-cas.3boxlabs.com/tcp/4012/wss/p2p/QmPHLQoWhK4CMPPgxGQxjNYEp1fMB8NPpoLaaR2VDMNbcr',
+    '/dns4/ipfs-clay-cas.3boxlabs.com/tcp/4012/wss/p2p/QmQotCKxiMWt935TyCBFTN23jaivxwrZ3uD58wNxeg5npi',
+    '/dns4/ipfs-clay.ceramic.network/tcp/4012/wss/p2p/QmQqq1oVkwpJdKNdiWvrftgWE2nov4AFGPBauZ2AakxmoR',
+    '/dns4/ipfs-clay.3boxlabs.com/tcp/4012/wss/p2p/QmQotCKxiMWt935TyCBFTN23jaivxwrZ3uD58wNxeg5npi'
+]
+
+export async function buildIpfsConnection(network: string, ipfsEndpoint?: string): Promise<IpfsApi>{
+    if (ipfsEndpoint) {
+        return ipfsClient({ url: ipfsEndpoint, ipld: { formats: [dagJoseFormat] } })
+    } else {
+        const ipfs = await IPFS.create({
+            ipld: {
+                formats: [dagJoseFormat]
+            },
+            libp2p: {
+                config: {
+                    dht: {
+                        enabled: true,
+                        clientMode: !IPFS_DHT_SERVER_MODE,
+                        randomWalk: false,
+                    },
+                },
+                pubsub: {
+                    enabled: true
+                },
+            },
+            config: {
+                Routing: {
+                    Type: IPFS_DHT_SERVER_MODE ? 'dhtserver' : 'dhtclient',
+                },
+                Bootstrap: BOOTSTRAP_LIST
+            }
+        })
+        const shouldConnect = network === "testnet-clay" || network === "mainnet"
+        if (shouldConnect) {
+            await Promise.all(BOOTSTRAP_LIST.map(async node => {
```

Comment:
Do we really need to await this? I don't think this is blocking for the rest of the code?

Reply:
It's controversial. If you do not wait, the daemon _certainly_ proceeds with an empty set of peers. If you wait, it leaves you with the "best effort" connectivity, which most probably is all right for application to proceed.